Musicca/app/src/main/java/com/example/musicca/adapters/QueueAdapter.java

```diff
@@ -100,23 +104,67 @@ protected void publishResults(CharSequence charSequence, FilterResults filterRes
         };
     }
 
+    // Access the data result passed to the activity here
+    // perform filtering through the filter dialog fragment
+    public void performMultiFiltering(String yearAfter, String yearBefore, String songTitle, String songArtist) {
+
+        List<Song> filteredList = new ArrayList<>();
+
+        int searchYearAfter;
+        int searchYearBefore;
+
+        if (TextUtils.equals(yearAfter, DEFAULT_STRING_AFTER)) {
+            searchYearAfter = 0;
+        } else {
+            searchYearAfter = Integer.valueOf(yearAfter);
+        }
+
+        if (TextUtils.equals(yearBefore, DEFAULT_STRING_BEFORE)){
+            searchYearBefore= 3000;
+        } else {
+            searchYearBefore = Integer.valueOf(yearBefore);
+        }
+
+
+        for (Song song : songsAll) {
+            if ((yearAfter == DEFAULT_STRING_AFTER || Integer.valueOf(song.getYear().toLowerCase()) >= searchYearAfter) &&
+                    (yearBefore == DEFAULT_STRING_BEFORE|| Integer.valueOf(song.getYear().toLowerCase()) <= searchYearBefore) &&
```

Comment:
Given you called `Integer.valueOf` on the year twice, you can make it a local variable to reuse.
`int songYear = Integer.valueOf(song.getYear().toLowerCase());`.